source/simulator/helper_functions.cc

```diff
@@ -1402,8 +1402,10 @@ namespace aspect
     Quadrature<dim> quadrature_formula(quadrature_points);
 
     // Quadrature rules only used for the numerical integration for better accuracy
-    const QGauss<dim> quadrature_formula_0 (advection_field.polynomial_degree(introspection)+1);
-
+    const Quadrature<dim> &quadrature_formula
+      = (advection_field.is_temperature() ?
+         introspection.quadratures.temperature :
+         introspection.quadratures.compositional_fields);
```

Comment:
It took me quite a while to understand that here we are not always dealing with compositional fields, and that the tests that fail are exactly those where the temperature and compositional fields use different polynomial degrees -- which seems to be the case only for melt tests, getting me completely onto the wrong track :-)

Anyway, with a bit of luck this will now test correctly.